src/internal/connector/onedrive/item_test.go

```diff
@@ -256,3 +258,158 @@ func (suite *ItemIntegrationSuite) TestDriveGetFolder() {
 		})
 	}
 }
+
+// Unit tests
+
+// mock
+type mockGetter struct {
+	GetFunc func(ctx context.Context, url string) (*http.Response, error)
+}
+
+func (m mockGetter) Get(
+	ctx context.Context,
+	url string,
+	headers map[string]string,
+) (*http.Response, error) {
+	return m.GetFunc(ctx, url)
+}
+
+type ItemUnitTestSuite struct {
+	tester.Suite
+}
+
+func TestItemUnitTestSuite(t *testing.T) {
+	suite.Run(t, &ItemUnitTestSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ItemUnitTestSuite) TestDownloadItem() {
+	testRc := ioutil.NopCloser(bytes.NewReader([]byte("test")))
+
+	table := []struct {
+		name          string
+		itemFunc      func() models.DriveItemable
+		GetFunc       func(ctx context.Context, url string) (*http.Response, error)
+		errorExpected bool
+		rcExpected    bool
```

Comment:
you can actually specify assert/require functions as part of the struct. For example, you could have the below types and then just give the function in the struct definitions and call the function in the test itself
```suggestion
		errorExpected require.ErrorAssertionFunc
		rcExpected    require.ValueAssertionFunc // single-arg assertion func
```

in struct definitions:
```go
{
  ...
  errorExpected: require.NoError // or use require.Error if there should be an error
  rcExpected: require.NotNil // or use require.Nil for no rc
}
```

calling in test:
```go
  test.errorExpected(t, err, clues.ToCore(err))
  test.rcExpected(t, rc)
```